### PR TITLE
Test metadata for all tests touched in a PR

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,8 +50,12 @@ test-metadata:
 test-metadata-changed:
 	tools/check_metadata $$(git diff --name-only | grep 'tests.*pm')
 
+.PHONY: test-metadata-merge
+test-metadata-merge:
+	tools/check_metadata $$(git diff --name-only FETCH_HEAD `git merge-base FETCH_HEAD master` | grep 'tests.*pm')
```
Comment (okurz) [inline Makefile]: Fails when there are no changes under tests.*pm

```bash
files=$(git diff --name-only FETCH_HEAD `git merge-base FETCH_HEAD master` | grep 'tests.*pm')
[ "$files" != "" ] && tools/check_metadata $files
```
?

Comment (coolo) [inline Makefile]: yeah, that's why I was testing it :)

**Comment by coolo** [comment]
works - direct pushes are ok, PRs require metadata for the changed tests.

**Comment by okurz** [comment]
LGTM

**Pull request merged**